x-pack/plugins/security_solution/server/endpoint/utils/action_list_helpers.ts

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from '@kbn/core/server';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import type { SearchRequest } from '@kbn/data-plugin/public';
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import type { TransportResult } from '@elastic/elasticsearch';
+import { AGENT_ACTIONS_RESULTS_INDEX } from '@kbn/fleet-plugin/common';
+import { EndpointError } from '../../../common/endpoint/errors';
+import type { SecuritySolutionRequestHandlerContext } from '../../types';
+import type {
+  EndpointAction,
+  LogsEndpointAction,
+  ActionListApiResponse,
+  EndpointActionResponse,
+  LogsEndpointActionResponse,
+} from '../../../common/endpoint/types';
+import {
+  ACTIONS_SEARCH_PAGE_SIZE,
+  ACTION_REQUEST_INDICES,
+  ACTION_RESPONSE_INDICES,
+} from '../services/actions/constants';
+import { getDateFilters, logsEndpointActionsRegex } from '../services/actions/utils';
+import { catchAndWrapError } from './wrap_errors';
+import { doesLogsEndpointActionsIndexExist } from './yes_no_data_stream';
+import { GetActionDetailsListParam } from '../services/actions/action_list';
+import { ENDPOINT_ACTIONS_INDEX } from '../../../common/endpoint/constants';
+
+const queryOptions = {
+  headers: {
+    'X-elastic-product-origin': 'fleet',
+  },
+  ignore: [404],
+};
+
+export const getTimeSortedActionDetails = (
+  data: ActionListApiResponse['data']
+): ActionListApiResponse['data'] => {
+  return data.sort((a, b) => (new Date(b.startedAt) > new Date(a.startedAt) ? 1 : -1));
+};
+
+export const getActionsForAgents = async ({
+  actionTypes,
+  context,
+  elasticAgentIds,
+  endDate,
+  from,
+  logger,
+  size,
+  startDate,
+  userIds,
+}: GetActionDetailsListParam): Promise<{
```

Comment:
can you define a type that is specific to this utility function. Is everything in `GetActionDetailsListParam` needed by this function?

Also,
as I stated in my prior comment, this method is named very speicifc to getting action only for certain set of Agents, which I would I would think that `elasticAgentIds` would be required and that array should hav a size of 1 at the very least.